src/registrar/tests/test_admin.py

```diff
@@ -153,6 +157,33 @@ def test_save_model_sends_approved_email(self):
         # Perform assertions on the mock call itself
         mock_client_instance.send_email.assert_called_once()
 
+    def test_save_model_sets_approved_domain(self):
```

Comment:
Add a test for the negative case. That if the domain status transition fails the domain is not linked to it. That part is critical and would break a lot of things. so good to ensure it is always working now that it is separated out.